athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v4/ObjectMapperFactoryV4.java

```diff
@@ -0,0 +1,367 @@
+import com.amazonaws.athena.connector.lambda.serde.v2.LambdaFunctionExceptionSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ListSchemasRequestSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ListSchemasResponseSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ListTablesRequestSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ListTablesResponseSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.MarkerSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.RangeSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ReadRecordsRequestSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ReadRecordsResponseSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.RemoteReadRecordsResponseSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.S3SpillLocationSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.SortedRangeSetSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.SpillLocationSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.SplitSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.TableNameSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.UserDefinedFunctionRequestSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.UserDefinedFunctionResponseSerDe;
+import com.amazonaws.athena.connector.lambda.serde.v2.ValueSetSerDe;
+import com.amazonaws.services.lambda.invoke.LambdaFunctionException;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.cfg.DeserializerFactoryConfig;
+import com.fasterxml.jackson.databind.cfg.SerializerFactoryConfig;
+import com.fasterxml.jackson.databind.deser.BeanDeserializerFactory;
+import com.fasterxml.jackson.databind.deser.DefaultDeserializationContext;
+import com.fasterxml.jackson.databind.deser.DeserializerFactory;
+import com.fasterxml.jackson.databind.deser.Deserializers;
+import com.fasterxml.jackson.databind.module.SimpleDeserializers;
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+import com.fasterxml.jackson.databind.ser.BeanSerializerFactory;
+import com.fasterxml.jackson.databind.ser.SerializerFactory;
+import com.fasterxml.jackson.databind.ser.Serializers;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+public class ObjectMapperFactoryV4
```

Comment:
Just noting this again that I'm very much against all of this rote code for doing "serdes". All of this would have been just auto generated if we just used protobuf IDL + its json encoding.